.travis.yml

```diff
@@ -1,27 +1,57 @@
-sudo: false
 language: python
 
 env:
-    - CONDA="python=2.7"
-    - CONDA="python=3.3"
-    - CONDA="python=3.4"
+  - CONDA_PYTHON_VERSION="2"
+  - CONDA_PYTHON_VERSION="3"
```

Comment:
> That said, the original configuration file did not seem to be testing all the Python versions.

How is that? We are testing from python 2.7 to Python 3.4 (3.5 now) skiping, on purpose 3.2 and 3.1.  I'd rather keep the current configuration then just declaring major versions.

> I followed instructions and recommendations made here: http://conda.pydata.org/docs/travis.html. 

I don't see major python versions those recommendations.  Also, conda is pretty fast in creating the ENV matrix for the testing, no need to download a minconda version for each travis python matrix.

> I did like the recommendation of using conda build inside the test script to catch any issues with the build scripts.

I disagree with that, but that is a decision for the authors.  However, if that is implemented you could try splitting Travis into a build and test component.  Here is an example: https://github.com/SciTools/iris/blob/master/.travis.yml#L14-L19